### PR TITLE
interfaces: add seccomp support for hooks.

### PR DESCRIPTION
**Pull request by kyrofa**
More progress on LP: [#1586465](https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1586465). Also add Spread tests to verify profile generation for both apps and hooks.

### CODE REVIEW

File: tests/seccomp-profiles/task.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  done
+execute: |
+  seccomp_profile_directory="/var/lib/snapd/seccomp/profiles"
+
+  echo "seccomp profiles are generated for apps"
+  sudo snap install ./basic-binaries_1.0_all.snap
+
+  for profile in snap.basic-binaries.block snap.basic-binaries.cat snap.basic-binaries.echo snap.basic-binaries.fail snap.basic-binaries.success
+  do
+    [ -f "$seccomp_profile_directory/$profile" ] || exit 1
```
Comment (niemeyer) [inline tests/seccomp-profiles/task.yaml]: This test is duplicating the whole logic just to add this additional line, it seems?  Can we just add this one line into the other test and rename it to security-profiles?

**Comment by niemeyer** [comment]
A note about test duplication and then LGTM.

File: tests/seccomp-profiles/task.yaml (hunk 1 above)
Comment (kyrofa) [inline tests/seccomp-profiles/task.yaml]: Yep, you're right. Done.

**Comment by kyrofa** [comment]
retest this please.

**Comment by mvo5** [comment]
:+1:

**Pull request merged**